packages/flutter_tools/lib/src/android/gradle_utils.dart

```diff
@@ -24,33 +24,43 @@ import 'android_sdk.dart';
 // In general, Flutter aims to default to the latest version.
 // However, this currently requires to migrate existing integration tests to the latest supported values.
 //
-// For more information about the latest version, check:
-// https://developer.android.com/studio/releases/gradle-plugin#updating-gradle
-// https://kotlinlang.org/docs/releases.html#release-details
+// Please see the README before changing any of these values.
 const String templateDefaultGradleVersion = '7.5';
 const String templateAndroidGradlePluginVersion = '7.3.0';
-const String templateDefaultGradleVersionForModule = '7.3.0';
+const String templateAndroidGradlePluginVersionForModule = '7.3.0';
 const String templateKotlinGradlePluginVersion = '1.7.10';
 
-// These versions should match the values in Flutter Gradle Plugin (FlutterExtension).
-// The Flutter Gradle plugin is only applied to app projects, and modules that are built from source
-// using (include_flutter.groovy).
-// The remaining projects are: plugins, and modules compiled as AARs. In modules, the ephemeral directory
-// `.android` is always regenerated after flutter pub get, so new versions are picked up after a
-// Flutter upgrade.
+// Please see the README before changing any of these values.
 const String compileSdkVersion = '33';
 const String minSdkVersion = '19';
 const String targetSdkVersion = '33';
 const String ndkVersion = '23.1.7779620';
 
+
+// Update these when new major versions of Java are supported by Gradle.
+// Source of truth: https://docs.gradle.org/current/userguide/compatibility.html
+const String oneMajorVersionHigherJavaVersion = '20';
+
 // Update this when new versions of Gradle come out including minor versions
+// and should correspond to the maximum Gradle version we test in CI.
 //
 // Supported here means supported by the tooling for
 // flutter analyze --suggestions and does not imply broader flutter support.
-const String _maxKnownAndSupportedGradleVersion = '8.0.2';
+const String maxKnownAndSupportedGradleVersion = '8.0.2';
+
 // Update this when new versions of AGP come out.
+//
+// Supported here means tooling is aware of this version's Java <-> AGP
+// compatibility.
 @visibleForTesting
-const String maxKnownAgpVersion = '8.1';
+const String maxKnownAndSupportedAgpVersion = '8.1';
+
+// Update this when new versions of AGP come out.
+const String maxKnownAgpVersion = '8.2';
```

Comment:
Since the start of this pr 8.3 has been released. https://developer.android.com/build/releases/gradle-plugin